periodic sync upstream KF to midstream ODH

Al-Pragliola opened the pull request:
superseeds https://github.com/opendatahub-io/model-registry/pull/276

- keep https://github.com/kubeflow/model-registry/pull/1323
- keep https://github.com/kubeflow/model-registry/pull/1303
- keep https://github.com/kubeflow/model-registry/pull/1329
- keep https://github.com/kubeflow/model-registry/pull/1331
- keep https://github.com/kubeflow/model-registry/pull/1334
- keep https://github.com/kubeflow/model-registry/pull/1336
- keep https://github.com/kubeflow/model-registry/pull/1332
- keep https://github.com/kubeflow/model-registry/pull/1322
- keep https://github.com/kubeflow/model-registry/pull/1338
- keep https://github.com/kubeflow/model-registry/pull/1339
- keep https://github.com/kubeflow/model-registry/pull/1349
- keep https://github.com/kubeflow/model-registry/pull/1348
- keep https://github.com/kubeflow/model-registry/pull/1345
- keep https://github.com/kubeflow/model-registry/pull/1347
- keep https://github.com/kubeflow/model-registry/pull/1352
- keep https://github.com/kubeflow/model-registry/pull/1355
- keep https://github.com/kubeflow/model-registry/pull/1354
- keep https://github.com/kubeflow/model-registry/pull/1353
- keep https://github.com/kubeflow/model-registry/pull/1358
- merge commit


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Improved error handling and validation for pagination and input parameters, enhancing security and robustness.
  * Expanded support for storing and updating model descriptions and documentation via the Python client and REST API.

* **Bug Fixes**
  * Structured error messages are now returned for malformed API requests, improving clarity for end-users.
  * Enhanced security for pagination queries to prevent injection vulnerabilities.

* **Refactor**
  * Updated build and deployment workflows for improved reliability and maintainability.
  * Updated Dockerfiles and dependency management for better consistency and compatibility.

* **Style**
  * Improved UI layout for model registry tables to enhance usability.

* **Documentation**
  * Updated README and diagrams to reflect current testing strategies and tooling.

* **Tests**
  * Replaced Robot Framework end-to-end tests with new Python-based tests, focusing on user story validation and error handling.
  * Added comprehensive unit tests for pagination and TLS configuration.

* **Chores**
  * Removed legacy Robot Framework test files and related resources.
  * Upgraded multiple dependencies for improved security and performance.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tarilabs commented:
I'm not entirely sure why a blank line in GHA workflow file is a formatting "issue" 🤔 , as you will find many examples on GitHub docs provided that way.

Besides that, you might want consider excluding some files from this scope, to make the build passing again.

isinyaaa commented:
codecov seems to be failing for no apparent reason, I'll wait a while before re-triggering.

tarilabs reviewed (approved):
As previously approved in https://github.com/opendatahub-io/model-registry/pull/285#issuecomment-1920666695 but I don't believe empty space in YAML for GHA is technically "an issue", given most examples from GH docs are designed that way. Given this PR is a prerequisite for other more priority work, I'd merge this asap.

lampajr reviewed (approved):
I agree that new lines in GHA yaml shouldn't be an issue, maybe just some config that can be tweaked (?)

btw it looks good to me!

Al-Pragliola committed:
chore: remove unused dep (go-enum) (#1323)

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

rsun19 committed:
fix: centered header buttons (#1303)

* centered header buttons

Signed-off-by: rsun19 <robertssun1234@gmail.com>

* removed unnecessary flex property

Signed-off-by: rsun19 <robertssun1234@gmail.com>

* added todo comment

Signed-off-by: rsun19 <robertssun1234@gmail.com>

---------

Signed-off-by: rsun19 <robertssun1234@gmail.com>

jonburdo committed:
chore: fix "Reviewers" URL in pull request template (#1329)

Signed-off-by: Jon Burdo <jon@jonburdo.com>

tarilabs committed:
feat: migrate remainder of User Story from Robot to pytest (and DocArtifact) (#1331)

* feat: migrate remainder of User Story from Robot to pytest (and DocArtifact)

- Migrate Robot Framework user stories to pytest with comprehensive test coverage:
  - Model name storage and validation
  - Model description storage and updates
  - Longer documentation storage via DocArtifact
- Enhance ModelVersion with registered_model_id field and improved documentation
- Add DocArtifact create() and update() methods with proper artifact type mapping
- Improve Makefile deployment with conditional image building (helpful locally)
- Add migration comments to existing Robot Framework tests per convention

This change completes the DocArtifact implementation and modernizes the test suite
by migrating user stories from Robot Framework to pytest for better maintainability
and as previously agreed and enacted.

With this PR, all Robot UserStor-ies are migrated to pytest

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

* linting

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

* implement code review feedback

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

---------

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

Al-Pragliola committed:
feat: remove linting/gen from dockerfile (#1334)

* feat: remove linting/gen from dockerfile

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix: missing checkout in action

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix: change prepare step

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* feat: make test target atomic

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* feat: make the jobs depend on prepare

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

---------

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

tarilabs committed:
fix: remove incorrect prepare workflow call in build-and-push-image.yml (#1336)

follow-up to https://github.com/kubeflow/model-registry/pull/1334

The build-and-push-image workflow was incorrectly calling the prepare.yml
as a step action instead of using it as a reusable workflow. This caused
the error "Can't find 'action.yml', 'action.yaml' or 'Dockerfile' under
.github/workflows/prepare.yml".

Example: https://github.com/kubeflow/model-registry/actions/runs/16379707719/job/46288348593

The prepare job is already correctly called at the top of the jobs section,
so the duplicate call in the steps was unnecessary and likely the cause
of the failure.

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

tarilabs committed:
feat: migrate "Regression" user stories from Robot to pytest  (#1332)

* feat: migrate "Regression" user stories from Robot to pytest

- Add structured error message tests for malformed model data (migrated from Robot)
- Test API returns proper error messages for malformed RegisteredModel and ModelVersion requests
  - for both `code` and `message`
- Clean ups/linting

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

* add comment to Robot

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

* linting

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

---------

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

mprahl committed:
Make the openapi client a separate Go module (#1322)

This will allow Kubeflow Pipelines to use the Go client directly without
needing to import all of Model Registry and inheriting its dependencies.

This introduces the concept of a Go workspace to allow multiple Go
modules to live in the same repo.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

tarilabs committed:
refactor: remove remaining Robot Framework after migration (#1338)

- Remove Robot Framework test infrastructure and files
- Update README.md to reflect KinD and Pytest for e2e testing
- Replace testing diagram with drawio version
- Remove robot-tests GitHub workflow

This concludes the migration of Robot to Pytest
by removing the remainder files and ci/GHA from the repo

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>

Al-Pragliola committed:
feat: address high code security alerts (#1339)

* feat: solve tls security issues

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix: solve unsafe int conversion

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix: pagination hardening

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix: check insecure cipher after parsing

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix: use sanitezed params in pagination ordering

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix: pagination always default

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

---------

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

dependabot[bot] committed:
build(deps): bump gorm.io/driver/postgres from 1.5.7 to 1.6.0 (#1349)

Bumps [gorm.io/driver/postgres](https://github.com/go-gorm/postgres) from 1.5.7 to 1.6.0.
- [Commits](https://github.com/go-gorm/postgres/compare/v1.5.7...v1.6.0)

---
updated-dependencies:
- dependency-name: gorm.io/driver/postgres
  dependency-version: 1.6.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/go-logr/logr from 1.4.2 to 1.4.3 (#1348)

Bumps [github.com/go-logr/logr](https://github.com/go-logr/logr) from 1.4.2 to 1.4.3.
- [Release notes](https://github.com/go-logr/logr/releases)
- [Changelog](https://github.com/go-logr/logr/blob/master/CHANGELOG.md)
- [Commits](https://github.com/go-logr/logr/compare/v1.4.2...v1.4.3)

---
updated-dependencies:
- dependency-name: github.com/go-logr/logr
  dependency-version: 1.4.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/testcontainers/testcontainers-go from 0.37.0 to 0.38.0 (#1345)

Bumps [github.com/testcontainers/testcontainers-go](https://github.com/testcontainers/testcontainers-go) from 0.37.0 to 0.38.0.
- [Release notes](https://github.com/testcontainers/testcontainers-go/releases)
- [Commits](https://github.com/testcontainers/testcontainers-go/compare/v0.37.0...v0.38.0)

---
updated-dependencies:
- dependency-name: github.com/testcontainers/testcontainers-go
  dependency-version: 0.38.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump k8s.io/apimachinery from 0.33.2 to 0.33.3 (#1347)

Bumps [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery) from 0.33.2 to 0.33.3.
- [Commits](https://github.com/kubernetes/apimachinery/compare/v0.33.2...v0.33.3)

---
updated-dependencies:
- dependency-name: k8s.io/apimachinery
  dependency-version: 0.33.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump github.com/testcontainers/testcontainers-go/modules/mysql from 0.37.0 to 0.38.0 (#1352)

Bumps [github.com/testcontainers/testcontainers-go/modules/mysql](https://github.com/testcontainers/testcontainers-go) from 0.37.0 to 0.38.0.
- [Release notes](https://github.com/testcontainers/testcontainers-go/releases)
- [Commits](https://github.com/testcontainers/testcontainers-go/compare/v0.37.0...v0.38.0)

---
updated-dependencies:
- dependency-name: github.com/testcontainers/testcontainers-go/modules/mysql
  dependency-version: 0.38.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Al-Pragliola committed:
chore: move to more general purpose tag for go 1.24 (#1355)

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

dependabot[bot] committed:
build(deps): bump k8s.io/api from 0.33.2 to 0.33.3 (#1354)

Bumps [k8s.io/api](https://github.com/kubernetes/api) from 0.33.2 to 0.33.3.
- [Commits](https://github.com/kubernetes/api/compare/v0.33.2...v0.33.3)

---
updated-dependencies:
- dependency-name: k8s.io/api
  dependency-version: 0.33.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.72.1 to 1.73.0 (#1353)

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.72.1 to 1.73.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.72.1...v1.73.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-version: 1.73.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Al-Pragliola committed:
fix: build issues post go.work file introduction (#1358)

* fix: add gorm-gen to the go workspace file

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* feat: make the gha run without a paths filter

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix: disable workspace for gorm-gen commands

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

* fix: bff ci build step

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

---------

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

Al-Pragliola committed:
Merge branch 'main' into al-pragliola-20250721-sync

tarilabs commented:
https://github.com/opendatahub-io/model-registry/pull/285#issue-3249568631

/lgtm
